### PR TITLE
Insectoids (Roachmen) II: Electric bugaloo

### PR DESCRIPTION
**Pull request by Buchse**
Alright, after testing and some help from some very friendly coders, I've got roach men working.

Insectoids completely ignore radiation, and take half toxin damage. 
However, in exchange they suffer 1.3x burn and brute damage, making roaches fairly fragile
Special thanks to
@ancientpower 
@DeityLink 
@boy2mantwicethefam 
:cl:
 * rscadd: Adds the Insectoid race, a flag that allows radiation to be ignored, the "insectoid" language, and a new subset of monkey. 
* rscadd: Fixed a conflict with diona hair.
* rscadd: Fixed the insectoid language by changing say key
* rscadd: Adds insectoid organs (eyes, heart, lungs) and adds them to the bioprinter. Insectoids now have random bug-sounding names.
Insectoid Front
insectoid back
insectoid side
insectoid organs
insectoid language
insectoid language don't speak

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by SonixApache** [comment]
roach memes are trash

**Comment by ghost** [comment]
sonix is trash and hates fun

File: code/modules/mob/living/carbon/monkey/roach.dm [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	mob_swap_flags = MONKEY|SLIME|SIMPLE_ANIMAL
+	mob_push_flags = MONKEY|SLIME|SIMPLE_ANIMAL|ALIEN
+
+	canWearClothes = 0
+	canWearHats = 1
+	canWearGlasses = 0
+	greaterform = "Insectoid"
+	languagetoadd = LANGUAGE_INSECT
+
+/mob/living/carbon/monkey/rock/passive_emote()
```
Comment (boy2mantwicethefam) [inline code/modules/mob/living/carbon/monkey/roach.dm]: fix this pls

File: code/modules/mob/new_player/sprite_accessories.dm [hunk 2]
```diff
@@ -508,9 +508,6 @@
 	icon_state = "hair_manbun"
 	gender = MALE
 
-/datum/sprite_accessory/hair/bald
```
Comment (MadmanMartian) [inline code/modules/mob/new_player/sprite_accessories.dm:511]: RIP Bald?

File: code/modules/mob/living/carbon/monkey/roach.dm (hunk 1 above)
Comment (Buchse) [reply code/modules/mob/living/carbon/monkey/roach.dm]: ah, hang on, I HAD fixed that

File: code/modules/mob/new_player/sprite_accessories.dm (hunk 2 above)
Comment (Buchse) [reply code/modules/mob/new_player/sprite_accessories.dm:511]: nah, that was a test copy I made while fiddling

File: code/modules/mob/living/carbon/monkey/roach.dm (hunk 1 above)
Comment (Buchse) [reply code/modules/mob/living/carbon/monkey/roach.dm]: done

File: code/modules/mob/new_player/sprite_accessories.dm (hunk 2 above)
Comment (boy2mantwicethefam) [reply code/modules/mob/new_player/sprite_accessories.dm:511]: bald is redefined above, so that's weird

**Review by boy2mantwicethefam — APPROVED** [review]

File: code/modules/mob/new_player/sprite_accessories.dm (hunk 2 above)
Comment (Buchse) [reply code/modules/mob/new_player/sprite_accessories.dm:511]: One of the tests I had done was to make a version of "bald" that was explicitly insectoid only, before I realized there was nothing wrong with the hair. Bald is still there, and works fine.

**Comment by Kharkov72** [comment]
I for one welcome our turkish overlords

**Comment by SonixApache** [comment]
i may hate fun but regardless of that i see no reason for "roach men" to exist on a space station

**Comment by boy2mantwicethefam** [comment]
roach people are okay in my books tbh, could've been far worse, roach people are on the upper scale of races we should have

**Comment by N3X15** [comment]
Avalitard should take note:  If you're willing to put in the effort, the maintainers are willing to look at adding it.  Whether it dies on GitHub or not due to the meme of emojiocracy is another question.

Spamming the thread just makes you look like a impotent 12-year-old.

**Review by N3X15 — APPROVED** [review]
Code looks good, but you should post some in-game pics (all sides) so we can evaluate the sprites.

**Comment by SonixApache** [comment]
@N3X15 please don't validate that "person"

**Comment by beesil** [comment]
Suggestion:
Make them neuter (like slimepeople/golems/mushrooms) so
1. They're more alien, and 
2. We don't end up with bug girl w*ifus

**Comment by Buchse** [comment]
I'll post some sprites once I'm able to connect back to BYOND and host a test serb

**Comment by DamianX** [comment]
Very lame, reskinned humans with minor stat differences and free shard-extinguisher combo availability.
Why do they have lungs, kidneys, appendices, livers? Why do they have human variants of all organs, including eyes?
It seems very low effort. I didn't even look at the sprites but I'm upvoting because it's content and therefore inherently good

**Comment by ghost** [comment]
> 
> 
> Very lame, reskinned humans with minor stat differences and free shard-extinguisher combo availability.
> Why do they have lungs, kidneys, appendices, livers? Why do they have human variants of all organs, including eyes?

Do you have any ideas to make them more unique? Most of those issues are because Buchse isn't much of a coder and was recommended to copypaste until it worked. I'm willing to code more complicated functionality for this if people have good suggestions.

Right now I have some ideas of my own but they involve giving insects a subspecies system which involves this passing first.

**Comment by DrSnips** [comment]
Yeah being 100% rad immune does open up the issue of taking the shard for walks around the station while looking like a fireman. For once I have no ideas on how to balance this.

**Comment by Buchse** [comment]
> Very lame, reskinned humans with minor stat differences and free shard-extinguisher combo availability.
> Why do they have lungs, kidneys, appendices, livers? Why do they have human variants of all organs, including eyes?
> It seems very low effort. I didn't even look at the sprites but I'm upvoting because it's content and therefore inherently good

I have some ideas on things to change to make them more alien, especially in regards to different organs for the future, my main concern was getting them working in the first place. The shard-dragging shouldn't be any larger of an issue than borgs being able to do it, in my opinion.

**Comment by SageOfLight** [comment]
> Suggestion:
> Make them neuter (like slimepeople/golems/mushrooms) so
> 
> 1. They're more alien, and
> 2. We don't end up with bug girl w*ifus

People are gonna do weird shit regardless of what the game tells them, we don't have actual genitals in the game after all.

**Comment by Intigracy** [comment]
Don't you put down the male character's two pixel long schlong as not being real, that's rude.

**Comment by calzilla1** [comment]
Make stomping do tripple damage

**Comment by ghost** [comment]
> i may hate fun but regardless of that i see no reason for "roach men" to exist on a space station

I think i can write a pretty believable fluff for em based on SCP-3035.

**Review by ShiftyRail — APPROVED** [review]
Looks ok, I think the `:y` key might already be taken, though.

**Comment by DrSnips** [comment]
I don't think we have any keys left for language/comms. Why does every race need a snowflake language again?

**Comment by Caboozles** [comment]
I’m hoping fluff-wise they’re even lower than vox.

**Comment by Buchse** [comment]
I've got the language working by changing the key to something unused

**Comment by DrSnips** [comment]
When do we cut less/unused race languages to free up some characters? Stuff like skrell or lizard would be a start.

**Comment by Buchse** [comment]
> When do we cut less/unused race languages to free up some characters? Stuff kil skrell or lizard would be a start.

Well, skrell was removed and it's language key taken by grey, but lizard should be ditched, in my opinion. Same for tajaran honest,y. It would free up the O and J keys for something else.

**Comment by strawberrytan** [comment]
if they had special rad immune organs you could put in humans, that would be neat for the extremely niche ghoul creation process

File: code/modules/mob/language.dm [hunk 3]
```diff
@@ -101,6 +101,16 @@
...
 	"SKRE","AHK","EHK","RAWK","KRA","AAA","EEE","KI","II","KRI","KA")
 
+/datum/language/insectoid
+	name = LANGUAGE_INSECT
+	desc = "A collection of disquieting vibrations and chittering sounds, the spoken tongue of insectoids. "
+	speech_verb = "chitters"
+	ask_verb = "clicks"
+	exclaim_verb = "hisses"
+	colour = "gutter"
+	key = "%"
```
Comment (boy2mantwicethefam) [inline code/modules/mob/language.dm:111]: 😱

Comment (MadmanMartian) [reply code/modules/mob/language.dm:111]: 😨

Comment (Buchse) [reply code/modules/mob/language.dm:111]: wait what's the issue with this?

**Comment by tuypo1** [comment]
As for shards maybe make them mostly rad immune but still take damage to some minor organ that they can survive without but will be less effective with it damaged, perhaps something that slows them down if they take too many rads

File: code/modules/organs/internal/lungs/lung.dm [hunk 4]
```diff
@@ -138,3 +138,14 @@
 		new /datum/lung_gas/waste(GAS_CARBON,         max_pp=10),
 		new /datum/lung_gas/sleep_agent(GAS_SLEEPING, min_giggle_pp=0.15, min_para_pp=1, min_sleep_pp=5),
 	)
+
+/datum/organ/internal/lungs/insectoid
+	name = "\improper Insectoid lungs"
```
Comment (Kurfursten) [inline code/modules/organs/internal/lungs/lung.dm:143]: You call these book lungs elsewhere, keep it consistent

**Comment by Kurfursten** [comment]
Just one minor nitpick, otherwise seems fine. Go ahead and post your pics

**Comment by AI-Is-Rogue** [comment]
oh no they're cute

**Comment by d3athrow** [comment]
I can't believe the turks have won

**Pull request merged**